### PR TITLE
Fix multipolygon geometry iterator.

### PR DESCRIPTION
**Pull request by bdice**
## Description
This PR fixes a bug in the multipolygon geometry iterator. The geometry iterator was returning part iterators by mistake, which masked an issue that we were patching out in rapids-cmake's CCCL. See https://github.com/rapidsai/rapids-cmake/pull/511.

With this fix, we can remove that patch from rapids-cmake's CCCL: https://github.com/rapidsai/rapids-cmake/pull/640

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cuspatial/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

File: cpp/include/cuspatial_test/vector_factories.cuh
Comment (bdice) [inline cpp/include/cuspatial_test/vector_factories.cuh:167]: Without these changes, I was seeing errors that no `operator<<` overload existed for `thrust::host_vector`.

File: cpp/include/cuspatial/range/multipolygon_range.cuh
Comment (bdice) [inline cpp/include/cuspatial/range/multipolygon_range.cuh:121]: This is the core bug: the `thrust::copy` call using the geometry offsets was getting the wrong size because the wrong iterator was being returned.

This only became visible in the one test case where a multipolygon contained more than one polygon.

File: cpp/tests/range/multipolygon_range_test.cu
Comment (bdice) [inline cpp/tests/range/multipolygon_range_test.cu:1072]: This was a typo that was masked by some kind of silent failure earlier in the tests, I think. The geometry offsets are `{0, 2}`. (I'm not entirely sure how this test ever passed.)

**Review by miscco — APPROVED** [review]

**Review by harrism — APPROVED** [review]
Thanks for fixing! @isVoid should review too.

**Comment by bdice** [comment]
Thanks for the reviews. There are some failing tests but I think they are related to #1400 and not this PR. I pinged @mroeschke to look into that PR, and I'll merge it into this one once it passes. Then we can remove the CCCL patches from cuDF and rapids-cmake.

File: cpp/include/cuspatial_test/vector_factories.cuh
Comment (trxcllnt) [reply cpp/include/cuspatial_test/vector_factories.cuh:167]: Did thrust remove this overload from host_vector?

**Review by trxcllnt — APPROVED** [review]

File: cpp/include/cuspatial_test/vector_factories.cuh
Comment (bdice) [reply cpp/include/cuspatial_test/vector_factories.cuh:167]: I wasn't aware that such an overload ever existed. I don't see that implemented: https://github.com/search?q=repo%3ANVIDIA%2Fcccl%20operator%3C%3C&type=code

**Pull request merged**
